APMrover2/AP_MotorsUGV.cpp

```diff
@@ -299,6 +299,22 @@ bool AP_MotorsUGV::output_test_pwm(motor_test_order motor_seq, float pwm)
     return true;
 }
 
+//  returns true if checks pass, false if they fail.  report should be true to send text messages to GCS
+bool AP_MotorsUGV::pre_arm_check(bool report) const
+{
+    // check if both regular and skid steering functions have been defined
+    if (SRV_Channels::function_assigned(SRV_Channel::k_throttleLeft) &&
```

Comment:
Surely it's a misconfiguration to define throttle left but not throttle right?

And a misconfiguration to not have both k_throttle and k_steering defined at the same time?

You could get some really weird effects with this current logic, I think; if the user has only one of the skid-steering outputs set, this `&&` stuff will be false so the prearm won't fail.

Reply:
I think what you're suggesting are additional checks beyond the one that I was trying to capture.

The hidden control problems occur when the AP_MotorsUGV::have_skid_steering() method returns true (i.e. outputs for both throttleLeft and throttleRight are configure) but we don't actually have any motors connected to those channels and instead are using the regular throttle and steering.  In these cases the vehicle appears to drive somewhat normally but actually the steering control is scaled incorrectly.  So from the user's point of view, it's a non-obvious error.

If the user only defined one output to use throttleLeft or throttleRight the vehicle simply wouldn't move.  If the user was using a regular style rover and didn't define one of throttle or steering then the vehicle would either not move or would be unable to steer.  So it's a fairly obvious that something is wrong.

Reply:
.. but i will add some more checks.

thanks for the review!

Reply:
Looks good!